lib/matplotlib/transforms.py

```diff
@@ -1340,24 +1340,32 @@ def contains_branch_seperately(self, other_transform):
 
     def __sub__(self, other):
         """
-        Return a transform stack which goes all the way down self's transform
-        stack, and then ascends back up other's stack. If it can, this is
-        optimised::
-
-            # normally
-            A - B == a + b.inverted()
-
-            # sometimes, when A contains the tree B there is no need to
-            # descend all the way down to the base of A (via B), instead we
-            # can just stop at B.
-
-            (A + B) - (B)^-1 == A
-
-            # similarly, when B contains tree A, we can avoid descending A at
-            # all, basically:
-            A - (A + B) == ((B + A) - A).inverted() or B^-1
-
-        For clarity, the result of ``(A + B) - B + B == (A + B)``.
+        Compose *self* with the inverse of *other*, cancelling identical terms
+        if any::
+
+            # In general:
+            A - B == A + B.inverted()
+            # (but see note regarding frozen transforms below).
+
+            # If A "ends with" B (i.e. A == A' + B for some A') we can cancel
+            # out B:
+            (A' + B) - B == A'
+
+            # Likewise, if B "starts with" A (B = A + B'), we can cancel out A:
+            A - (A + B') == B'.inverted() == B'^-1
+
+        Cancellation (rather than naively composing ``A`` with ``B.inverted()``
```

Comment:
```suggestion
        Cancellation (rather than naively composing ``A`` with ``B.inverted()``)
```
(but it needs wrapping then)